src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,102 @@
+import { FC, useMemo, useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import cn from 'classnames';
+import { TodoList } from '../TodoList/TodoList';
+import { Todo, FILTERS } from '../../types/types';
+
+const LINKS = [
+  {
+    to: '/',
+    recentFilter: FILTERS.all,
+  },
+  {
+    to: '/active',
+    recentFilter: FILTERS.active,
+  },
+  {
+    to: '/completed',
+    recentFilter: FILTERS.completed,
+  },
+];
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  activeIds: number[];
+  handleRemoveTodo: (id: number) => void;
+  handleClearCompleted: () => void;
+};
+
+export const Filter: FC<Props> = ({
+  todos,
+  tempTodo,
+  activeIds,
+  handleRemoveTodo,
+  handleClearCompleted,
+}) => {
+  const [filter, setFilter] = useState<FILTERS>(FILTERS.all);
```

Comment:
Instead of using FILTERS directly in useState, consider using the initialFilter as a prop with a default value of FILTERS.all. This way, the initial filter can be controlled from the parent component.